src/arch/amdgpu/vega/insts/flat.cc

```diff
@@ -2227,5 +2467,133 @@ namespace VegaISA
     {
         atomicComplete<VecOperandF64, VecElemF64>(gpuDynInst);
     } // completeAcc
+    // --- Inst_FLAT__FLAT_ATOMIC_PK_ADD_BF16 class methods ---
+
+    Inst_FLAT__FLAT_ATOMIC_PK_ADD_BF16::Inst_FLAT__FLAT_ATOMIC_PK_ADD_BF16(
+        InFmt_FLAT *iFmt)
+        : Inst_FLAT(iFmt, "flat_atomic_pk_add_bf16")
+    {
+        setFlag(AtomicPkAddBF16);
+
+        // MI300 spec: "Float atomics must set SC[0]=0 (no return value)."
+        panic_if(instData.GLC, "Saw float atomic with return set!");
```

Comment:
is there a way for us to transform GLC into SC[0]?  I guess not with backwards support?